presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -2782,6 +2782,15 @@ public void testJoinWithScalarSubqueryToBeExecutedAsPostJoinFilterWithEmptyInner
     }
 
     @Test
+    public void testInUncorrelatedSubquery()
+            throws Exception
+    {
+        assertQuery(
+                "SELECT CASE WHEN false THEN 1 IN (VALUES 2) END",
```

Comment:
It is not obvious that this test tests the rule you have just added.

Does below query reproduces this problem also?
```
SELECT (VALUES 3) FROM VALUES (1, 2);
```
If so you could also handle correlated queries:

```
SELECT (VALUES a + 1) FROM VALUES (1, 2) t(a);
```